From 6f414862735fa7c6a344e95088444a31de60ae28 Mon Sep 17 00:00:00 2001
Message-Id: <6f414862735fa7c6a344e95088444a31de60ae28.1425610844.git.chang-joon.lee@intel.com>
In-Reply-To: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
References: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Akashdeep Sharma <akashdeep.sharma@intel.com>
Date: Mon, 16 Feb 2015 23:52:25 +0530
Subject: [PATCH 17/27] SQUASHME! [VPG]: video/adf: NULL Pipe check for HDMI
 Audio

Added NULL Pipe check in HDMI LPE Audio register and set up APIs.
These APIs should return error in case no HDMI encoder/ pipe is
enumerated. Else, it will result in a crash.

Also, moved audio init call after extcon device registration in
hdmi_pipe_init.

SQUASHME! - This patch should be squashed into the following existing patch:
	Author: Akashdeep Sharma <akashdeep.sharma@intel.com>
	Date:   Mon Feb 2 10:10:22 2015 +0530
	Change-Id: I50a0a3240115b86a5931472e72901173472e72f1
	MUST_REBASE [VPG]: video/adf: Audio Hotplug and DIP

Issue: GMINL-5532
Change-Id: Ic72364d72a2fc4ae908330b4f2ea9e54701c7ba4
Signed-off-by: Akashdeep Sharma <akashdeep.sharma@intel.com>
---
 .../adf/intel/core/common/hdmi/gen_hdmi_audio.c    |   10 ++++++++++
 .../adf/intel/core/common/hdmi/gen_hdmi_pipe.c     |    4 ++--
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_audio.c b/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_audio.c
index 42d912b..1ed3095 100644
--- a/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_audio.c
+++ b/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_audio.c
@@ -217,6 +217,11 @@ int adf_hdmi_audio_setup(
 
 	pr_debug("ADF: HDMI:%s\n", __func__);
 
+	if (!pipe) {
+		pr_err("ADF: HDMI:%s HDMI Pipe is NULL\n", __func__);
+		return -ENODEV;
+	}
+
 	reg_ops->hdmi_audio_get_register_base =
 			(hdmi_audio_reg_ops.hdmi_audio_get_register_base);
 	reg_ops->hdmi_audio_read_register =
@@ -243,6 +248,11 @@ int adf_hdmi_audio_register(
 
 	pr_debug("ADF: HDMI:%s\n", __func__);
 
+	if (!pipe) {
+		pr_err("ADF: HDMI:%s HDMI Pipe is NULL\n", __func__);
+		return -ENODEV;
+	}
+
 	hdmi_priv.had_pvt_data = had_data;
 	hdmi_priv.had_interface = driver;
 
diff --git a/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_pipe.c b/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_pipe.c
index b05bfe3..2c8cafb 100644
--- a/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_pipe.c
+++ b/drivers/video/adf/intel/core/common/hdmi/gen_hdmi_pipe.c
@@ -213,13 +213,13 @@ int hdmi_pipe_init(struct hdmi_pipe *pipe,
 		goto fail_pipe;
 	}
 
-	adf_hdmi_audio_init(pipe);
-
 #ifdef CONFIG_EXTCON
 	pipe->hotplug_switch.name = "hdmi";
 	extcon_dev_register(&pipe->hotplug_switch);
 #endif
 
+	adf_hdmi_audio_init(pipe);
+
 	/*
 	 * Configure HDMI
 	 * This function can probe HDMI and try to do a modeset
-- 
1.7.9.5

